### PR TITLE
cmd/telemetrygen: Add headers to gRPC metadata for logs

### PR DESCRIPTION
**Pull request by metalmatze**
**Description:** <Describe what has changed.>
So far, sending (HTTP) headers along with gRPC requests hasn't been possible.
For that reason, sending bearer tokens for authentication didn't work either.

This is fixed now.

Updated version of #30082
cc @mx-psi

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Thanks for reopening, LGTM!

**Comment by metalmatze** [comment]
Rebased against the latest `main`. It was quite a bit outdated by now.

File: .chloggen/telemetrygen-logs-grpc-metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: telemetrygen
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add headers to gRPC metadata for logs
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: 32668
```
Comment (songy23) [inline .chloggen/telemetrygen-logs-grpc-metadata.yaml:13]:
```suggestion
issues: [32668]
```

**Review by songy23 — APPROVED** [review]

File: .chloggen/telemetrygen-logs-grpc-metadata.yaml (hunk 1 above)
Comment (metalmatze) [reply .chloggen/telemetrygen-logs-grpc-metadata.yaml]: Thanks!

**Pull request merged**